### PR TITLE
Session type filter visualizations

### LINKED ISSUE #2227: Add Filter Option for The Session Type Visualization
A visualization such as the one pasted below could use a filtering option I believe since it is hard to determine what's what. It's hard to hover over those bars to know which one to click to drill down. 

image

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #2227 

this adds the filter box to the course/session-type visualizations that already exists for course/instructors.

### CODE REVIEW

File: addon/components/course-visualize-instructors.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { restartableTask } from 'ember-concurrency';
 export default class CourseVisualizeInstructorsComponent extends Component {
   @service iliosConfig;
   @tracked academicYearCrossesCalendarYearBoundaries = false;
+  @tracked name;
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/course-visualize-instructors.js:9]: i put this in, it was missing. we're using a setter on that attribute in the component template.

[the remark this answers is not in the document]
Comment (jrjohnson) [reply addon/components/course-visualize-instructors.js:9]: ✅

**Review by jrjohnson — APPROVED** [review]

File: addon/components/visualizer-course-session-types.js [hunk 2]
```diff
@@ -18,6 +18,22 @@ export default class VisualizerCourseSessionTypes extends Component {
...
   }
 
+  get filteredData() {
+    if (!this.data) {
+      return [];
+    }
+
+    let data = this.data;
+    if (this.args.filter) {
+      const exp = new RegExp(this.args.filter, 'gi');
```
Comment (jrjohnson) [inline addon/components/visualizer-course-session-types.js:28]: Late breaking comment... this probably needs `import { cleanQuery } from 'ilios-common/utils/query-utils';` to ensure the query string is regex safe before using it.

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: addon/components/visualizer-course-session-types.js (hunk 2 above)
Comment (stopfstedt) [reply addon/components/visualizer-course-session-types.js]: good catch. fixed.

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**